extensions/git/src/model.ts

```diff
@@ -769,17 +769,17 @@ export class Model implements IBranchProtectionProviderRegistry, IRemoteSourcePu
 	registerBranchProtectionProvider(root: Uri, provider: BranchProtectionProvider): Disposable {
 		const providerDisposables: Disposable[] = [];
 
-		this.branchProtectionProviders.set(root, (this.branchProtectionProviders.get(root) ?? new Set()).add(provider));
+		this.branchProtectionProviders.set(root.fsPath, (this.branchProtectionProviders.get(root.fsPath) ?? new Set()).add(provider));
```

Comment:
Dunno if it's an issue here, but in other places where we use a map keyed by a uri, we use `toString` for the key because fsPath is losing some properties

Reply:
Thanks! Fixed!